### ISSUE TITLE
TestNG 7.4.0 assertSame / assertNotSame checks for Equality

### ISSUE THREAD

**Issue by jensGiehl** [cbeust/testng#2527]
### TestNG Version
7.4.0

### Expected behavior
`assertSame` should check that two objects refer to the same object. 
`assertNotSame` should check that two objects do not refer to the same objects.

### Actual behavior
Since 7.4.0 (Commit: 841ae94) `assertSame` (and `assertNotSame`) use `Objects.equals`, so it's checks for Identity or Equality.

### Is the issue reproductible on runner?

- [ ] Shell
- [X] Maven
- [ ] Gradle
- [ ] Ant
- [X] Eclipse
- [ ] IntelliJ
- [ ] NetBeans

### Test case sample
```
@Test(expectedExceptions = AssertionError.class)
	public void shouldThrowAnAssertionErrorBecauseObjectsAreNotIdentical() throws Exception {
		MyObject obj1 = new MyObject("Hello world");
		MyObject obj2 = new MyObject("Hello world");

		assertSame(obj1, obj2);
	}

	@Test
	public void objectsShouldNotBeIdentical() throws Exception {
		MyObject obj1 = new MyObject("Hello world");
		MyObject obj2 = new MyObject("Hello world");

		assertNotSame(obj1, obj2);
	}

	public static class MyObject {

		private String content;

		public MyObject(String content) {
			super();
			this.content = content;
		}

		@Override
		public int hashCode() {
			return Objects.hash(content);
		}

		@Override
		public boolean equals(Object obj) {
			if (this == obj) {
				return true;
			}
			if (!(obj instanceof MyObject)) {
				return false;
			}
			MyObject other = (MyObject) obj;
			return Objects.equals(content, other.content);
		}
	}
```

**Comment by juherr** [comment]
Hi, thanks for the report.

Would you like to try to fix it?

**Comment by jensGiehl** [comment]
Sure. Will take care of it tonight.

**Comment by jensGiehl** [comment]
Duplication of #2487.

Fixed with c89fbb6. 

But the commit was done after the 7.4.0 release, but in the `CHANGES.txt` it is under 7.4.0 release
